crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1460,26 +1460,14 @@ where
 /// Once the [RpcModule] is built via [RpcModuleBuilder] the servers can be started, See also
 /// [ServerBuilder::build] and [Server::start](jsonrpsee::server::Server::start).
 #[derive(Default, Debug)]
-pub struct RpcServerConfig {
-    /// Configs for JSON-RPC Http.
-    http_server_config: Option<ServerBuilder<Identity, Identity>>,
-    /// Allowed CORS Domains for http
-    http_cors_domains: Option<String>,
-    /// Address where to bind the http server to
-    http_addr: Option<SocketAddr>,
-    /// Configs for WS server
-    ws_server_config: Option<ServerBuilder<Identity, Identity>>,
-    /// Allowed CORS Domains for ws.
-    ws_cors_domains: Option<String>,
-    /// Address where to bind the ws server to
-    ws_addr: Option<SocketAddr>,
-    /// Configs for JSON-RPC IPC server
-    ipc_server_config: Option<IpcServerBuilder<Identity, Identity>>,
```

Comment:
we should keep all of those and instead use the additional HTTP and rpc fields for _all_ ServerBuilders that were installed, this makes it to configure these layers for all servers

Reply:
would that be in `launch_rpc_servers`? 

i added some kind of sketch of using it with an `.option_layer(config.rpc.clone())` but that was a little bit of a stab in the dark, i'm trying to pick back up the thread of this pr rn